### PR TITLE
Add cron support to CLI

### PR DESCRIPTION
**Pull request by yiminc**
I have to use client's master branch as the new release is not cut yet.

### CODE REVIEW

File: tools/cli/commands.go [hunk 1]
```diff
@@ -559,7 +560,7 @@ func StartWorkflow(c *cli.Context) {
 	tcCtx, cancel := newContext()
 	defer cancel()
 
-	resp, err := serviceClient.StartWorkflowExecution(tcCtx, &s.StartWorkflowExecutionRequest{
+	startRequest := &s.StartWorkflowExecutionRequest{
```
Comment (longquanzheng) [inline tools/cli/commands.go]: If you use the serverFrontendClient, then you don't have to update the client repo IDLs to have CLI. Refering this: https://github.com/uber/cadence/blob/master/tools/cli/commands.go#L1176

File: Gopkg.lock [hunk 2]
```diff
@@ -526,7 +526,7 @@
   version = "v1.3.2"
 
 [[projects]]
-  digest = "1:fd58696ccbfea94bb5773204f42c11416197587a1d22da198c251bcee859e572"
+  digest = "1:5e70aa639271a54e9ed87875b9f7e495255df54b1a64c7d4a338e33b44ee33b7"
```
Comment (longquanzheng) [inline Gopkg.lock]: You also need to stash this change since you don't change Gopkg.toml anymore

Comment (yiminc) [reply Gopkg.lock]: done

File: tools/cli/workflow.go [hunk 3]
```diff
@@ -134,6 +134,17 @@ func newWorkflowCommands() []cli.Command {
 					Value: defaultDecisionTimeoutInSeconds,
 					Usage: "Decision task start to close timeout in seconds",
 				},
+				cli.StringFlag{
```
Comment (vancexu) [inline tools/cli/workflow.go]: Could you also add this to `RunWorkflow`? There are duplicate code from RunWorkflow and `StartWorkflow` which would be good to refactor.

File: tools/cli/commands.go [hunk 4]
```diff
@@ -613,22 +620,27 @@ func RunWorkflow(c *cli.Context) {
...
 			Name: common.StringPtr(taskList),
 		},
 		Input:                               []byte(input),
 		ExecutionStartToCloseTimeoutSeconds: common.Int32Ptr(int32(et)),
 		TaskStartToCloseTimeoutSeconds:      common.Int32Ptr(int32(dt)),
 		Identity:                            common.StringPtr(getCliIdentity()),
 		WorkflowIdReusePolicy:               reusePolicy,
-	})
+	}
+	if c.IsSet(FlagCronSchedule) {
```
Comment (vancexu) [inline tools/cli/commands.go]: You also need to add this flag to to Run command, like the one you already added for Start command.

**Pull request merged**